nemo_skills/mcp/servers/brave_tool.py

```diff
@@ -0,0 +1,102 @@
+import argparse
+import logging
+import os
+from dataclasses import dataclass
+from typing import Annotated
+
+import httpx
+from mcp import StdioServerParameters
+from mcp.server.fastmcp import FastMCP
+from pydantic import Field
+
+from nemo_skills.mcp.clients import MCPStdioClient
+from nemo_skills.mcp.tool_providers import MCPClientTool
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ExecutionResult:
+    error: str | None = None
+    result: str | None = None
+
+
+mcp = FastMCP(name="brave")
+
+# Populated from CLI args in main()
+BRAVE_API_KEY: str | None = None
+
+
+## See docs https://api-dashboard.search.brave.com/app/documentation/summarizer-search/get-started
+@mcp.tool()
+async def summarize(
+    query: Annotated[str, Field(description="Search query.")],
+) -> ExecutionResult:
+    """Get a summary of search results from the web using Brave."""
+
+    if not BRAVE_API_KEY:
+        return {"error": "Missing Brave API key"}
```

Comment:
By the way, I think for errors like this for both this tool and python tool (and in general for any tools) we should do a hard fail and stop the job. These errors aren't recoverable and if users asked us to use a particular tool, we don't want to keep running it and just returning errors on every request. We should only soft-fail errors that could be recoverable, not missing api key or connection to sandbox kind of things